lib/lib-utils/src/main/java/org/sagebionetworks/util/json/JavaJSONUtil.java

```diff
@@ -84,6 +94,84 @@ public static <T> List<T> readFromJSON(Class<? extends T> clazz, JSONArray array
 		return list;
 	}
 
+	/**
+	 * Read a list of objects directly from the provided Stream
+	 * @param <T>
+	 * @param clazz
+	 * @param reader
+	 * @return
+	 */
+	public static <T> List<T> readFromJSON(Class<? extends T> clazz, Reader reader) {
+		ValidateArgument.required(reader, "reader");
+		ValidateArgument.required(clazz, "clazz");
+		Map<String, Object> typeMap = Arrays.stream(clazz.getDeclaredFields())
+				.collect(Collectors.toMap(f -> f.getName(), f -> f.getType()));
+		try {
+			List<T> list = new ArrayList<>();
+			try (JsonReader stream = new JsonReader(reader)) {
```

Comment:
Rather than using the gson library, we can use the JSONTokener(Reader) to read the stream. See JSONTokenener.nextClean() (e.g. starts with '[') and JSONTOkener.nextValue() (to read the next JSONObject).